pycuda/reduction.py

```diff
@@ -267,7 +268,11 @@ def __call__(self, *args, **kwargs):
                 macroblock_size = block_count*self.block_size
                 seq_count = (sz + macroblock_size - 1) // macroblock_size
 
-            if block_count == 1:
+            if block_count == 1 and out is not None:
+                assert out.dtype == self.dtype_out, \
```

Comment:
Please use `if`, not `assert`

Reply:
OK, switched to `if` & `raise ValueError`